Bugfix: When download files have zero length, now a 404 error code is…

KevinFHartmann opened the pull request:
… returned.

The resulting JSON now looks like this:
{
    "timestamp": 1490813376262,
    "status": 404,
    "error": "Not Found",
    "exception": "org.icgc.dcc.download.server.endpoint.NotFoundException",
    "message": "Job has no files to download",
    "path": "/downloads"
}
instead of this:
{
    "timestamp": 1490730306364,
    "status": 500,
    "error": "Internal Server Error",
    "exception": "java.lang.IllegalArgumentException",
    "message": "No message available",
    "path": "/downloads"
}

The server log will now contain a warning message that looks like this:

2017-03-29 14:49:36,260 [http-nio-9090-exec-6] WARN  o.i.d.d.s.u.Responses - In GzipStreamer.checkArguments(), job has no files to download

instead of an error message like this:

2016-10-27 10:26:42,546 [http-nio-8443-exec-5] ERROR o.a.c.c.C.[.[.[.[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processi
ng failed; nested exception is java.lang.IllegalArgumentException] with root cause
java.lang.IllegalArgumentException: null
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:77)
        at org.icgc.dcc.download.server.io.GzipStreamer.checkArguments(GzipStreamer.java:187)
        at org.icgc.dcc.download.server.io.GzipStreamer.<init>(GzipStreamer.java:84)

KevinFHartmann committed:
Bugfix: When download files have zero length, now a 404 error code is returned.

The resulting JSON now looks like this:
{
    "timestamp": 1490813376262,
    "status": 404,
    "error": "Not Found",
    "exception": "org.icgc.dcc.download.server.endpoint.NotFoundException",
    "message": "Job has no files to download",
    "path": "/downloads"
}
instead of this:
{
    "timestamp": 1490730306364,
    "status": 500,
    "error": "Internal Server Error",
    "exception": "java.lang.IllegalArgumentException",
    "message": "No message available",
    "path": "/downloads"
}

The server log will now contain a warning message that looks like this:

2017-03-29 14:49:36,260 [http-nio-9090-exec-6] WARN  o.i.d.d.s.u.Responses - In GzipStreamer.checkArguments(), job has no files to download

instead of an error message like this:

2016-10-27 10:26:42,546 [http-nio-8443-exec-5] ERROR o.a.c.c.C.[.[.[.[dispatcherServlet] - Servlet.service() for servlet [dispatcherServlet] in context with path [] threw exception [Request processi
ng failed; nested exception is java.lang.IllegalArgumentException] with root cause
java.lang.IllegalArgumentException: null
        at com.google.common.base.Preconditions.checkArgument(Preconditions.java:77)
        at org.icgc.dcc.download.server.io.GzipStreamer.checkArguments(GzipStreamer.java:187)
        at org.icgc.dcc.download.server.io.GzipStreamer.<init>(GzipStreamer.java:84)

dcc-jenkins commented:
Ready for review. 	- Type "ok to test" to initiate a PR build. 	- Type "test this please" to retest. 	- Type "add to whitelist" to always build on your PRs.	See https://wiki.jenkins-ci.org/display/JENKINS/GitHub+pull+request+builder+plugin

andricDu commented:
add to whitelist

andricDu commented on dcc-download-server/src/main/java/org/icgc/dcc/download/server/io/GzipStreamer.java:
```diff
@@ -184,7 +184,10 @@ private long getCurrentHeaderLength() {
   private void checkArguments() {
-    checkArgument(!downloadFiles.isEmpty());
+    if (downloadFiles.isEmpty()) {
+      Responses.throwNotFoundException("Job has no files to download",
```
Use a static import for `throwNotFoundException`

Should be `ctrl` + `shift` + `m` in eclipse if I recall correctly.

andricDu commented on dcc-download-server/src/main/java/org/icgc/dcc/download/server/io/GzipStreamer.java:
```diff
@@ -105,9 +106,8 @@ public boolean hasNext() {
-    val nextEntryName = fileNames.containsKey(downloadDataType) ?
-        fileNames.get(downloadDataType) :
-        getFileName(downloadDataType, empty()) + ".tsv.gz";
+    val nextEntryName = fileNames.containsKey(downloadDataType) ? fileNames
```
something happened to the formatting here. Put `fileNames` on the next line.

andricDu commented:
ok to test

KevinFHartmann committed:
Code Review Fixes:
  -- Fixed formatting of line 108
  -- Now uses a static import for throwNotFoundException

KevinFHartmann committed:
Formatting: Fixed formatting issues with GzipStreamer.java, and updated eclipse formatter settings.

Apparently, the Java code formatter in Eclipse 4.5 has been completely rewritten; with changes to how line wrapping is handled.

See https://bugs.eclipse.org/bugs/show_bug.cgi?id=372801 for a description of how Eclipse formatting changed for the 'Never join wrapped lines' option: it looks like it's saying that you can now only manually insert a line break if the formatter could have wrapped the line at that point (unless you tell the formatter to ignore formatting for that section with that ).

andricDu commented on dcc-download-server/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -97,11 +98,13 @@ org.eclipse.jdt.core.formatter.alignment_for_arguments_in_qualified_allocation_e
 org.eclipse.jdt.core.formatter.alignment_for_binary_expression=16
 org.eclipse.jdt.core.formatter.alignment_for_compact_if=0
-org.eclipse.jdt.core.formatter.alignment_for_conditional_expression=0
+org.eclipse.jdt.core.formatter.alignment_for_conditional_expression=80
```
👍
